dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpages/business/render/HTMLPageAssetRenderedAPIImplIntegretionTest.java

```diff
@@ -0,0 +1,418 @@
+     * When The host_id request's parameter is set, but the user does not have permission over it
+     * Should return a right {@link PageView}
+     * @throws DotDataException
+     * @throws DotSecurityException
+     * @throws InterruptedException
+     */
+    @Test(expected = DotSecurityException.class)
+    public void shouldUseHostIdParameterAndThrowDotSecurityException() throws DotDataException, DotSecurityException {
+        init();
+        final PageMode pageMode = PageMode.WORKING;
+        when(request.getParameter("host_id")).thenReturn(this.host.getIdentifier());
+        when(request.getAttribute(com.liferay.portal.util.WebKeys.USER)).thenReturn(user);
+        when(request.getAttribute(WebKeys.CURRENT_HOST)).thenReturn(host);
+
+        final HTMLPageAssetRenderedAPIImpl htmlPageAssetRenderedAPIImpl = new HTMLPageAssetRenderedAPIImpl();
+
+        htmlPageAssetRenderedAPIImpl.getPageRendered(request, response, user, htmlPageAsset.getURI(), pageMode);
+    }
+
+    /**
+     * Method to test: {@link HTMLPageAssetRenderedAPIImpl#getPageRendered(PageContext, HttpServletRequest, HttpServletResponse)
+     * When The {@link Host#HOST_VELOCITY_VAR_NAME} request's parameter is set, but the user does not have permission over it
+     * Should return a right {@link PageView}
+     * @throws DotDataException
+     * @throws DotSecurityException
+     * @throws InterruptedException
+     */
+    @Test(expected = DotSecurityException.class)
+    public void shouldUseHostNameParameterAndThrowDotSecurityException() throws DotDataException, DotSecurityException {
+        init();
+        final PageMode pageMode = PageMode.WORKING;
+        when(session.getAttribute( com.dotmarketing.util.WebKeys.CMS_SELECTED_HOST_ID )).thenReturn(host.getIdentifier());
+        when(request.getAttribute(com.liferay.portal.util.WebKeys.USER)).thenReturn(user);
+        when(request.getAttribute(WebKeys.CURRENT_HOST)).thenReturn(host);
+
+        final HTMLPageAssetRenderedAPIImpl htmlPageAssetRenderedAPIImpl = new HTMLPageAssetRenderedAPIImpl();
+
+        htmlPageAssetRenderedAPIImpl.getPageRendered(request, response, user, htmlPageAsset.getURI(), pageMode);
+    }
+
+    /**
+     * Method to test: {@link HTMLPageAssetRenderedAPIImpl#getPageRendered(PageContext, HttpServletRequest, HttpServletResponse)
+     * When The session host is set, but the user does not have permission over it
+     * Should return a right {@link PageView}
+     * @throws DotDataException
+     * @throws DotSecurityException
+     * @throws InterruptedException
+     */
+    @Test(expected = DotSecurityException.class)
+    public void shouldUseSessionHostAndThrowDotSecurityException() throws DotDataException, DotSecurityException {
+        init();
+        final PageMode pageMode = PageMode.WORKING;
+        when(session.getAttribute( com.dotmarketing.util.WebKeys.CMS_SELECTED_HOST_ID )).thenReturn(host.getIdentifier());
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unnecessary use of fully qualified name 'com.dotmarketing.util.WebKeys.CMS_SELECTED_HOST_ID' due to existing import 'com.dotmarketing.util.WebKeys'](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4718001)